distributed/diagnostics/nvml.py

```diff
@@ -134,29 +134,64 @@ def _pynvml_handles():
     count = device_get_count()
     if NVML_STATE == NVMLState.DISABLED_PYNVML_NOT_AVAILABLE:
         raise RuntimeError("NVML monitoring requires PyNVML and NVML to be installed")
-    elif NVML_STATE == NVMLState.DISABLED_LIBRARY_NOT_FOUND:
+    if NVML_STATE == NVMLState.DISABLED_LIBRARY_NOT_FOUND:
         raise RuntimeError("PyNVML is installed, but NVML is not")
-    elif NVML_STATE == NVMLState.DISABLED_WSL_INSUFFICIENT_DRIVER:
+    if NVML_STATE == NVMLState.DISABLED_WSL_INSUFFICIENT_DRIVER:
         raise RuntimeError(
             "Outdated NVIDIA drivers for WSL, please upgrade to "
             f"{MINIMUM_WSL_VERSION} or newer"
         )
-    elif NVML_STATE == NVMLState.DISABLED_CONFIG:
+    if NVML_STATE == NVMLState.DISABLED_CONFIG:
         raise RuntimeError(
             "PyNVML monitoring disabled by 'distributed.diagnostics.nvml' "
             "config setting"
         )
-    elif count == 0:
+    if count == 0:
         raise RuntimeError("No GPUs available")
-    else:
-        try:
-            gpu_idx = next(
-                map(int, os.environ.get("CUDA_VISIBLE_DEVICES", "").split(","))
+
+    device = 0
+    cuda_visible_devices = os.environ.get("CUDA_VISIBLE_DEVICES", "")
```

Comment:
I'm not entirely sure about how to treat `CUDA_VISIBLE_DEVICES` if it's empty. Does that mean that we see no devices or that the environment variable is unset and we see all of them?

Reply:
An empty string `CUDA_VISIBLE_DEVICES` means no devices are visible, whereas an unset `CUDA_VISIBLE_DEVICES` means all devices available are visible.

I'm not sure I understand what you're trying to do here specifically. Just above in line 149 we raise an exception if no devices are found, so I'm assuming you're trying to do something different than handle that case.